spec/firebase_app_distribution_api_client_spec.rb

```diff
@@ -128,7 +127,99 @@
   end
 
   describe '#upload' do
-    # Empty for now
+    let(:max_polling_retries) { Fastlane::Client::FirebaseAppDistributionApiClient::MAX_POLLING_RETRIES }
+    let(:polling_interval_seconds) { Fastlane::Client::FirebaseAppDistributionApiClient::POLLING_INTERVAL_SECONDS }
+    let(:upload_status_response_success) do
+      UploadStatusResponse.new(
+        { status: "SUCCESS",
+          release: { id: "release_id" } }
+      )
+    end
+    let(:upload_status_response_in_progress) do
+      UploadStatusResponse.new(
+        { status: "IN_PROGRESS",
+          release: {} }
+      )
+    end
+    let(:upload_status_response_error) do
+      UploadStatusResponse.new(
+        { status: "ERROR",
+          release: {} }
+      )
+    end
+
+    it 'skips the upload step if the binary has already been uploaded' do
+      expect(api_client).to receive(:get_upload_token)
```

Comment:
It looks like every spec is setting this expectation. Consider placing it in a before filter.

Reply:
Good catch!